### DISCUSSION TITLE
Oh my. I just ran osu lazer 2020 in Raspberry pi 4

### DISCUSSION THREAD

**Discussion by SpyrosTheBoss** [ppy/osu#24682] [Q&A]
Peppy tell me what you think
 https://drive.google.com/file/d/1CxyjbBkTgIbS9iQudVsRgeandn2ggUz_/view?usp=drivesdk

**Comment by peppy** [comment]
Have you tried with a more recent release? It would probably run a lot better.

**Reply by SpyrosTheBoss** [reply]
it sadly doesn't even boot up with new releases. it just crashes

**Reply by GYKgamer** [reply]
Hey, when you say crashes, do you mean it just doesn't turn on? I've been trying to build osu!lazer for the raspberry pi and i've gotten stuck where I can't turn on the game, it just doesn't open
![image](https://github.com/ppy/osu/assets/41442793/e21ac52d-2051-483e-a7f2-51a1a758e9f1)

**Reply by GYKgamer** [reply]
After a bit of work I got latest version successfully built 👀 
![image](https://github.com/ppy/osu/assets/41442793/5772601e-72fc-44d4-ae1c-8f4d89c555c0)

**Reply by ljasdf** [reply]
how are you able to do this @GYKgamer? i tried installing the box64 libraries that it errors out at, and it fixes nothing :(

**Reply by GYKgamer** [reply]
That's the neat part, I built osu for the CPU's architecture. No need for box64 / box86

**Reply by GYKgamer** [reply]
I did write a tutorial when I did this, but I do apologise for it as it's a bit jank.
https://github.com/GYKgamer/osu-raspberry-pi/blob/369a8b955724d14bba74c6536bf2b47ee12fa5f2/How%20to%20install.txt

**Reply by ljasdf** [reply]
thanks for the help, I have been looking for someone who has been able to get osu on an rpi(with or without box86/64), without luck, and I finally found one :) also thanks for the quick response. I don't get why osu doesn't just build it for arm and put it on their website, but it is what it is. I also might be able to get genuine osu running in the browser, most likely slowly af, if I'm able to get it built for x86 with cheerpx (browser, clientside, js, x86 Linux compiler). It would be cool af if osu had the whole thing on their website for people to do without downloading, but I doubt they would accept the idea 😭

**Reply by GYKgamer** [reply]
The arm part is probably because there may just be a time when there's some obscure proprietary thing that can't be built for arm/arm64, and arm64 windows can run x86 bit programs (x64 on windows 11) and i think osu has apple silicone support. Linux is just there as x86_64 only. But yeah peppy could definitely get arm builds done if he wanted too. Also a genuine osu on browser could work, but what's the point? You can't upload files if I remember correctly, only open them and then they import into osu (though I haven't tried that) also webosu already exists, which while unofficial is still really good (I've used it)

**Reply by 320kg** [reply]
> That's the neat part, I built osu for the CPU's architecture. No need for box64 / box86

...then why does your video have "Using native..." messages?

**Reply by GYKgamer** [reply]
What video? I've not made a video on this before, though now that you mention it I should. Also the screenshot also doesn't mention any "native usage" messages, and I know which one you're on about; the ones that show up on box64/86

**Reply by 320kg** [reply]
> What video?

My bad, I thought you are the discussion creator.

Anyways, can this be built on Alpine/pmOS? I imagine it wouldn't have score submission or multiplayer, however.

**Reply by GYKgamer** [reply]
It probably could, though you most likely would have to change the instructions to fit alpine, as for score submission or multiplayer, I wouldn't see why it wouldn't work, after all it is the actual game.

**Reply by 320kg** [reply]
Well, building from source in the first place results in the version being reported as 0.0.0 and lots of online functionality disabled

**Reply by GYKgamer** [reply]
In that case, unless you can code the online functionality back in, then no you couldn't sadly.
